core/src/main/java/org/hisp/dhis/android/core/D2Configuration.java

```diff
@@ -63,6 +67,9 @@ public abstract class D2Configuration {
     @NonNull
     public abstract List<Interceptor> networkInterceptors();
 
+    @NonNull
+    public abstract List<ClientPlugin<Unit>> networkplugins();
```

Comment:
Wouldn't be better to set use `Any` instead of `Unit` for the type? We would have `ClientPlugin<Any>`.

Checking the ClientPlugin interface, it seems it is not restricted to any type. If we use `Unit`, we restrict the plugins to have type `Unit`, which I think could have some implications in the configuration they might have